packages/_site/src/pages/components/dialog/index.tsx

```diff
@@ -0,0 +1,423 @@
+	ElementRef<typeof DialogPrimitive.Overlay>,
+	ComponentPropsWithoutRef<typeof DialogPrimitive.Overlay> & ExtendProps
+>(({ className, extend, ...props }, ref) => (
+	<DialogPrimitive.Overlay
+		ref={ref}
+		className={className}
+		{...stylex.props(styles.overlay, extend)}
+		{...props}
+	/>
+));
+DialogOverlay.displayName = DialogPrimitive.Overlay.displayName;
+
+const DialogContent = forwardRef<
+	ElementRef<typeof DialogPrimitive.Content>,
+	ComponentPropsWithoutRef<typeof DialogPrimitive.Content> & ExtendProps & CloseExtendProps
+>(({ className, extend, closeExtend, children, ...props }, ref) => (
+	<DialogPortal>
+		<DialogOverlay />
+		<DialogPrimitive.Content
+			ref={ref}
+			className={className}
+			{...stylex.props(styles.content, extend)}
+			{...props}
+		>
+			{children}
+
+			<DialogPrimitive.Close {...stylex.props(styles.close, closeExtend)}>
+				<svg
+					xmlns="http://www.w3.org/2000/svg"
+					viewBox="0 0 24 24"
+					fill="var(--text-color)"
+					width={"1.25rem"}
+					height={"1.25rem"}
+				>
+					<title>Close</title>
+					<path d="M11.9997 10.5865L16.9495 5.63672L18.3637 7.05093L13.4139 12.0007L18.3637 16.9504L16.9495 18.3646L11.9997 13.4149L7.04996 18.3646L5.63574 16.9504L10.5855 12.0007L5.63574 7.05093L7.04996 5.63672L11.9997 10.5865Z" />
+				</svg>
+			</DialogPrimitive.Close>
+		</DialogPrimitive.Content>
+	</DialogPortal>
+));
+DialogContent.displayName = DialogPrimitive.Content.displayName;
+
+const DialogHeader = ({
+	className,
+	extend,
+	...props
+}: HTMLAttributes<HTMLDivElement> & ExtendProps) => (
+	<div
+		className={className}
+		{...stylex.props(styles.header, extend)}
+		{...props}
+	/>
+);
+DialogHeader.displayName = "DialogHeader";
+
+const DialogFooter = ({
+	className,
+	extend,
+	...props
+}: HTMLAttributes<HTMLDivElement> & ExtendProps) => (
+	<div
+		className={className}
+		{...stylex.props(styles.footer, extend)}
+		{...props}
+	/>
+);
+DialogFooter.displayName = "DialogFooter";
+
+const DialogTitle = forwardRef<
+	ElementRef<typeof DialogPrimitive.Title>,
+	ComponentPropsWithoutRef<typeof DialogPrimitive.Title> & ExtendProps
+>(({ className, extend, ...props }, ref) => (
+	<DialogPrimitive.Title
+		ref={ref}
+		className={className}
+		{...stylex.props(styles.title, extend)}
+		{...props}
+	/>
+));
+DialogTitle.displayName = DialogPrimitive.Title.displayName;
+
+const DialogDescription = forwardRef<
+	ElementRef<typeof DialogPrimitive.Description>,
+	ComponentPropsWithoutRef<typeof DialogPrimitive.Description> & ExtendProps
+>(({ className, extend, ...props }, ref) => (
+	<DialogPrimitive.Description
+		ref={ref}
+		className={className}
+		{...stylex.props(styles.description, extend)}
+		{...props}
+	/>
+));
+DialogDescription.displayName = DialogPrimitive.Description.displayName;
+
+export {
+	Dialog,
+	DialogPortal,
+	DialogOverlay,
+	DialogTrigger,
+	DialogClose,
+	DialogContent,
+	DialogHeader,
+	DialogFooter,
+	DialogTitle,
+	DialogDescription,
+};
+`}
```

Comment:
I added a new prop `url` to `code block`. You can use `url={{"https://raw.githubusercontent.com/clearfeld/control-ui/main/packages/dialog/lib/index.tsx"}}` instead of the `code` prop, to auto get the latest code of the component.